### PR TITLE
GoogleSignIn updates

### PR DESCRIPTION
**Pull request by tvolkert**
* Switch to use GoogleSignIn CocoaPod instead of Google
    * Read client ID manually from `GoogleService-Info.plist`,
      and don't support passing the client ID from Dart anymore.
    
    https://github.com/flutter/flutter/issues/10819

### CODE REVIEW

**Comment by tvolkert** [comment]
Note: not bumping the version yet, as we'd still like to fix https://github.com/flutter/flutter/issues/10688 before releasing another version

**Comment by collinjackson** [comment]
Since this is a breaking change, we might want to bump the version to 0.1.0

File: packages/google_sign_in/lib/google_sign_in.dart [hunk 1]
```diff
@@ -125,12 +117,7 @@ class GoogleSignIn {
   /// The [hostedDomain] argument specifies a hosted domain restriction. By
   /// setting this, sign in will be restricted to accounts of the user in the
   /// specified domain. By default, the list of accounts will not be restricted.
-  ///
-  /// The [clientId] argument specifies the ID that was assigned to your app
-  /// when it was registered with Google. It is optional; if unspecified, it
-  /// will be deduced based on your project's metadata (e.g.
-  /// `GoogleServices-Info.plist` on iOS).
-  GoogleSignIn({this.scopes, this.hostedDomain, this.clientId});
+  GoogleSignIn({this.scopes, this.hostedDomain});
```
Comment (mehmetf) [inline packages/google_sign_in/lib/google_sign_in.dart]: Is there a deprecation path for external users?

We should probably at least put a comment here that ClientId has been deprecated and refer to readme for instructions on how to create the config file.

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m [hunk 2]
```diff
@@ -51,16 +51,12 @@ - (instancetype)init {
...
 - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   if ([call.method isEqualToString:@"init"]) {
-    NSError *error;
-    NSString *clientId = call.arguments[@"clientId"];
-    if (clientId && ![clientId isEqual:[NSNull null]]) {
-      [GIDSignIn sharedInstance].clientID = call.arguments[@"clientId"];
-    } else {
-      [[GGLContext sharedInstance] configureWithError:&error];
-    }
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"GoogleService-Info" ofType:@"plist"];
```
Comment (mehmetf) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: This crashes if file is not found right? We should probably return an informative platformexception instead.

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m [hunk 3]
```diff
@@ -51,16 +51,12 @@ - (instancetype)init {
...
+    NSMutableDictionary *plist = [[NSMutableDictionary alloc] initWithContentsOfFile:path];
+    [GIDSignIn sharedInstance].clientID = plist[@"CLIENT_ID"];
```
Comment (mehmetf) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Add comment to indicate where this const is coming from. If some time in the future this breaks due to a key rename, we know where to look to fix it.

File: packages/google_sign_in/lib/google_sign_in.dart (hunk 1 above)
Comment (tvolkert) [inline packages/google_sign_in/lib/google_sign_in.dart]: The current released version (`0.0.6`) doesn't have the clientId API in Dart - it was added last Friday and hasn't been released yet.  So we're good :-)

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m (hunk 2 above)
Comment (tvolkert) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Done

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m (hunk 3 above)
Comment (tvolkert) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Done

**Pull request merged**